File: 강의/24.08.06 - objc/2. ObjcPlusNumber/ObjcPlusNumber/ViewController.m
//
//  ViewController.m
//  ObjcPlusNumber
//
//  Created by 김종혁 on 8/6/24.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *numberLabel;

// count 변수 선언 ( 포인터 )
@property NSNumber *count;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // count에다가 0을 할당
    self.count = [NSNumber numberWithInt: 0];
    
    // updateLabel 호출
    [self updateLabel];
}

- (void)updateLabel {
    // value에다가 정수형 count값을 할당
    int value = [self.count intValue];
    
    // value를 문자열로 변환하여 text에 할당
    NSString *text = [[NSString alloc] initWithFormat:@"%d", value];
    
    // IBOutlet numberLabel을 text로 셋팅
    [self.numberLabel setText:text];
}

- (IBAction)plusOne:(id)sender {
    int value = [self.count intValue];
    value += 1;
    self.count = [NSNumber numberWithInt: value];
    
    [self updateLabel];
}

- (IBAction)plusFive:(id)sender {
    int value = [self.count intValue];
    value += 5;
    self.count = [NSNumber numberWithInt: value];
    [self updateLabel];
}

- (IBAction)plusTen:(id)sender {
    int value = [self.count intValue];
    value += 10;
    self.count = [NSNumber numberWithInt: value];
    [self updateLabel];
}

- (IBAction)resetBtn:(id)sender {
    int value = [self.count intValue];
    value = 0;
    self.count = [NSNumber numberWithInt: value];
    [self updateLabel];
}





@end

File: 24.06.04/NewCounterApp/NewCounterApp/ViewController.swift
//
//  ViewController.swift
//  NewCounterApp
//
//  Created by 김종혁 on 6/4/24.
//

import UIKit

class ViewController: UIViewController {
    
    var number: Int = 0
    
    @IBOutlet weak var numberLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        numberLabel.text = "\(number)"
    }
    
    @IBAction func minusOne(_ sender: Any) {
        minus()
    }
    @IBAction func plusOne(_ sender: Any) {
        plus(1)
    }
    @IBAction func plusFive(_ sender: Any) {
        plus(5)
    }
    @IBAction func plusTen(_ sender: Any) {
        plus(10)
    }
    @IBAction func resetButton(_ sender: Any) {
        plus(0)
    }
    
    func plus(_ count : Int){
        number += count
        
        if count == 0 {
            number = 0
        }
        numberLabel.text = "\(number)"
    }
    
    func minus(){
        if number > 0 {
            number -=  1
        } else {
            number = 0
        }
        numberLabel.text = "\(number)"
    }
    
}
